test/Marketplace.test.js

```diff
@@ -623,13 +644,37 @@ describe("Marketplace", function () {
         .withArgs(requestId(request))
     })
 
-    it("withdraws to the client", async function () {
+    it("withdraw rest of funds for finished requests", async function () {
+      await waitUntilStarted(marketplace, request, proof, token)
+      await waitUntilFinished(marketplace, requestId(request))
+
+      switchAccount(client)
+      const startBalance = await token.balanceOf(client.address)
+      await marketplace.withdrawFunds(slot.request)
+      const endBalance = await token.balanceOf(client.address)
+      expect(endBalance - startBalance).to.lt(
+        request.expiry * request.ask.reward
+      )
+    })
+
+    it("withdraws to the client when request is cancelled", async function () {
```

Comment:
Should there also be a test checking that cancelled requests pay out for slots that were filled before cancellation?